tests/_models/test_package.py

```diff
@@ -19,34 +20,61 @@ def sort_map_values(m: dict[str, Any]) -> dict[str, Any]:
 
 def test_guess_version_setuptools(mocker: MockerFixture) -> None:
     mocker.patch("pipdeptree._models.package.version", side_effect=PackageNotFoundError)
-    result = ReqPackage(mocker.MagicMock(key="setuptools")).installed_version
+    r = MagicMock()
+    r.name = "setuptools"
+    result = ReqPackage(r).installed_version
     assert result == "?"
 
 
 def test_dist_package_render_as_root() -> None:
-    foo = Mock(key="foo", project_name="foo", version="20.4.1")
+    foo = Mock(metadata={"Name": "foo"}, version="20.4.1")
     dp = DistPackage(foo)
     is_frozen = False
     assert dp.render_as_root(frozen=is_frozen) == "foo==20.4.1"
 
 
 def test_dist_package_render_as_branch() -> None:
-    foo = Mock(key="foo", project_name="foo", version="20.4.1")
-    bar = Mock(key="bar", project_name="bar", version="4.1.0")
-    bar_req = Mock(key="bar", project_name="bar", version="4.1.0", specs=[(">=", "4.0")])
+    foo = Mock(metadata={"Name": "foo"}, version="20.4.1")
+    bar = Mock(metadata={"Name": "bar"}, version="4.1.0")
+    bar_req = MagicMock(version="4.1.0", specifier=[">=4.0"])
+    bar_req.name = "bar"
     rp = ReqPackage(bar_req, dist=bar)
     dp = DistPackage(foo).as_parent_of(rp)
     is_frozen = False
     assert dp.render_as_branch(frozen=is_frozen) == "foo==20.4.1 [requires: bar>=4.0]"
 
 
+def test_dist_package_render_as_root_with_frozen() -> None:
+    json_text = '{"dir_info": {"editable": true}, "url": "file:///A/B/foo"}'
+    foo = Mock(metadata={"Name": "foo"}, version="20.4.1")
+    foo.read_text = Mock(return_value=json_text)
+    dp = DistPackage(foo)
+    is_frozen = True
+    expect = "# Editable install with no version control (foo===20.4.1)\n-e /A/B/foo"
+    assert dp.render_as_root(frozen=is_frozen) == expect
+
+
+def test_dist_package_as_frozen_repr(tmp_path: Path, monkeypatch: pytest.MonkeyPatch) -> None:
```

Comment:
You're testing `Package.as_frozen_repr()`, so let's have the test name reflect this.

Reply:
Done